aoc/d3/d3_test.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import "testing"
+
+// ensure the strings given get split in two and returned as
+// the first and second items in a slice
+func TestSplitString(t *testing.T) {
+	tests := []struct {
+		input   string
+		output1 string
+		output2 string
+	}{
+		{
+			input:   "abcd",
+			output1: "ab",
+			output2: "cd",
+		},
+		{
+			input:   "vJrwpWtwJgWrhcsFMMfFFhFp",
+			output1: "vJrwpWtwJgWr",
+			output2: "hcsFMMfFFhFp",
+		},
+		{
+			input:   "jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL",
+			output1: "jqHRNqRjqzjGDLGL",
+			output2: "rsFMfFZSrLrFZsSL",
+		},
+	}
+
+	for _, test := range tests {
+		result := SplitString(test.input)
+		if result[0] != test.output1 {
+			t.Errorf("Got %s, expected %s", result, test.output1)
+		}
+		if result[1] != test.output2 {
+			t.Errorf("Got %s, expected %s", result, test.output2)
+		}
+	}
```

Comment:
So, for every one of these cases, I'd probably run them inside a `t.Run()` to make them [subtests](https://pkg.go.dev/testing#T.Run). Otherwise if any of these cases fails it's reported as an entire test failure.
I also think you can use a single `output` as a `[]string` as well and use the great https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff. For example:
```go
import (
	"fmt"

	"github.com/google/go-cmp/cmp"
)

func main() {
	fmt.Println(cmp.Diff([]string{"hi", "helo"}, []string{"hi", "hello"}))
}
```

which returns
```
  []string{
  	"hi",
- 	"helo",
+ 	"hello",
  }

```

Reply:
I didn't know about subtests, very good use of goroutines! Test errors are clearer too, and I like that it forces to you name all your subtests. I've changed all tests to use subtests as they're all parametrised. And I'm not sure if it's related but running the tests went from 3s to 2s haha

Also for the output slice checking, I wasn't sure if that was possible, however I've now found [reflect.DeepEqual](https://pkg.go.dev/reflect#DeepEqual) and it fit exactly for this case. Changed in  https://github.com/lucasrijllart/advent-of-code-2022/pull/1/commits/85b685bb038e687cf47423e10f0b78c4b5ba021a

Reply:
2s!?!? Try changing to using bufio, that seems waaaaay too long